### ISSUE TITLE
[FR] Support "launch" request for debugging configuration

### ISSUE THREAD

**Issue by mike-lischke** [harold-b/vscode-duktape-debug#24]
Currently only the "attach" request setting is accepted for debugging a duktape application. It would however be much more convenient to have the debugger also launch the application to debug. Constantly having to manually start the application before starting a debugging run is tiring.

**Comment by willi1s** [comment]
Every debug target will be different, e.g. for the command line you need to run duk.exe with some flags, for embedded systems you may need to copy files and then maybe even run from a menu.

There is an option though, create a [task](https://code.visualstudio.com/docs/editor/tasks) that runs the code on your target, and then in the debug configuration set [preLaunchTask](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations) this should launch the app and then run the attach debug

**Comment by mike-lischke** [comment]
Debug targets might be different, but you usually can launch them e.g. by script, which must be run in a terminal and that's something the debug extension should be able to do. For more complex setup you still have the attach option. But I'll give the pre launch task a try. Thanks.

**Comment by harold-b** [comment]
> It would however be much more convenient to have the debugger also launch the application to debug.

I agree. And though it's on my list, I haven't had the time yet, sadly. Hopefully I can get to it sooner than later.

**Comment by mike-lischke** [comment]
@harold-b sorry for increasing the pressure here, but can you please give that problem a higher priority. Currently I have to start my project 100 times a day (because I'm working on tests which only need a few seconds to run, I fix them and re-run). And every time I start it I have to do it twice (because of the blocking prelaunch task) and have to dismiss the error afterwards. That's highly annyoing and very unproductive. Please consider adding a `launch` request type asap. Many thanks!
